src/app/main.rs

```diff
@@ -542,6 +543,10 @@ where
                     dst_route_layer,
                 ))
                 .buffer_pending(max_in_flight, DispatchDeadline::extract)
+                // Route to a balancer based on the ConcreteDst extention metadata
+                // if it has been set by the profiles router.
+                .layer(concrete_dst_router::layer(concrete_dst_router::Config::new("concrete dst", capacity, max_idle_age)))
```

Comment:
Thoughts on putting this in `Builder` in `app/svc.rs` similar to `buffer_pending` right below? I'm a little surprised rustfmt is not adding line breaks, but it seems like something to consider.

Reply:
I think removing the `capacity` and `max_idle_age` (as @olix0r suggested in https://github.com/linkerd/linkerd2-proxy/pull/270#discussion_r294905445) from the `concrete_dst_router::Config` type would help a bit. 

If we removed those fields from `Config`, we would only take a name:
```rust
  .layer(concrete_dst_router::layer(concrete_dst_router::Config::new("concrete dst")))
```
In that case, we could probably just pass the name directly to `layer` and have _it_ construct the `Config`:
```rust
  .layer(concrete_dst_router::layer("concrete dst"))
```
but we may want to add more configs to `Config` later, so 🤷‍♀